src/psyclone/tests/domain/lfric/lfric_config_test.py

```diff
@@ -130,10 +138,55 @@ def test_entry_not_bool(tmpdir, option):
         assert "Not a boolean: tree" in str(err.value)
 
 
+@pytest.mark.parametrize("option", ["NUM_ANY_SPACE",
+                                    "NUM_ANY_DISCONTINUOUS_SPACE"])
+def test_entry_not_int(tmpdir, option):
+    ''' Check that we raise an error if the value of any options expecting
+    an integer value is not int() with base 10. '''
+    content = re.sub(r"^{0} = .*$".format(option),
+                     "{0} = false".format(option),
+                     _CONFIG_CONTENT,
+                     flags=re.MULTILINE)
+    config_file = tmpdir.join("config_dyn")
+    with config_file.open(mode="w") as new_cfg:
+        new_cfg.write(content)
+        new_cfg.close()
+        config = Config()
+        with pytest.raises(ConfigurationError) as err:
+            config.load(config_file=str(config_file))
+
+        assert "error while parsing {0}".format(option) in str(err.value)
+        assert ("invalid literal for int() with base 10: 'false'"
+                in str(err.value))
+
+
+def test_invalid_fortran_datatype(tmpdir):
+    ''' Check that we raise an error if we supply an invalid Fortran
+    datatype in the configuration file.
+
+    '''
+    content = re.sub("real, integer, logical",
+                     "real, integrity, logical",
+                     _CONFIG_CONTENT)
+    config_file = tmpdir.join("config_dyn")
+    with config_file.open(mode="w") as new_cfg:
+        new_cfg.write(content)
+        new_cfg.close()
+        config = Config()
+        with pytest.raises(ConfigurationError) as err:
+            config.load(config_file=str(config_file))
+
+        assert ("Invalid Fortran datatype found in the \'[dynamo0.3]\' "
+                "section 'supported_fortran_datatypes'" in str(err.value))
+        assert ("Supported Fortran datatypes are: [\'real\', \'integer\', "
+                "\'logical\']." in str(err.value))
+
+
 def test_invalid_default_kind(tmpdir):
-    ''' Check that we raise an error if we supply an invalid datatype or
-    kind (precision) in the configuration file '''
+    ''' Check that we raise an error if we supply an invalid datatype or kind
+    (precision) in the 'default_kind' section of the configuration file.
```

Comment:
default_kind is not a section in the config file, it is a variable in the dynamo0.3 section.

Reply:
It is a list in the `[dynamo0.3]` section so I clarified the test description.